src/dali_executor/dali_executor.cc

```diff
@@ -21,28 +21,56 @@
 // SOFTWARE.
 
 #include "src/dali_executor/dali_executor.h"
-
 #include "src/dali_executor/utils/dali.h"
 
 namespace triton { namespace backend { namespace dali {
 
-template<bool owns>
-void DaliExecutor::SetupInputs(const std::vector<IODescr<owns>>& inputs) {
+void DaliExecutor::SetupInputs(const std::vector<IDescr>& inputs) {
   assert(!inputs.empty());
-  int batch_size = inputs[0].shape.num_samples();
+  int batch_size = inputs[0].meta.shape.num_samples();
   for (size_t i = 1; i < inputs.size(); ++i) {
-    assert(inputs[i].shape.num_samples() == batch_size &&
+    assert(inputs[i].meta.shape.num_samples() == batch_size &&
            "All inputs should have equal batch size.");
   }
+  std::vector<IDescr> c_inputs{};
   for (auto& inp : inputs) {
-    assert(inp.shape.num_elements() * dali_type_size(inp.type) <= inp.buffer.size());
-    pipeline_.SetInput(inp.buffer.data(), inp.name.c_str(), inp.device, inp.type, inp.shape);
+    size_t inp_size = inp.meta.shape.num_elements() * dali_type_size(inp.meta.type);
+    if (inp.buffers.size() == 1) {
+      assert(inp_size <= inp.buffers[0].size);
+      c_inputs.push_back(inp);
+    } else {
+      c_inputs.push_back(ScheduleInputCopy(inp));
```

Comment:
I would add a comment that you are merging requests into one continuous buffer.

Reply:
done